src/main/java/com/mojang/brigadier/CommandDispatcher.java

```diff
@@ -523,31 +493,28 @@ private String getSmartUsage(final CommandNode<S> node, final S source, final bo
                 final String redirect = node.getRedirect() == root ? "..." : "-> " + node.getRedirect().getUsageText();
                 return self + ARGUMENT_SEPARATOR + redirect;
             } else {
-                final Collection<CommandNode<S>> children = node.getChildren().stream().filter(c -> c.canUse(source)).collect(Collectors.toList());
+                final List<CommandNode<S>> children = new ArrayList<>();
+                for (final CommandNode<S> child : node.getChildren()) {
+                    if (child.canUse(source)) children.add(child);
+                }
                 if (children.size() == 1) {
-                    final String usage = getSmartUsage(children.iterator().next(), source, childOptional, childOptional);
-                    if (usage != null) {
-                        return self + ARGUMENT_SEPARATOR + usage;
-                    }
+                    final String usage = getSmartUsage(children.get(0), source, childOptional, childOptional);
+                    if (usage != null) return self + ARGUMENT_SEPARATOR + usage;
                 } else if (children.size() > 1) {
-                    final Set<String> childUsage = new LinkedHashSet<>();
+                    final List<String> childUsage = new ArrayList<>();
```

Comment:
If you want to make this even more effecient, you can use a LinkedList since there is no random access.

Reply:
Rodger that, thanks for the feedback on making this more efficient!